lib/utilities.bash

```diff
@@ -47,9 +47,10 @@ function _bash-it-get-component-type-from-path() {
 #
 #
 function _bash-it-array-contains-element() {
-	local e
-	for e in "${@:2}"; do
-		[[ "$e" == "$1" ]] && return 0
+	local e element="${1?}"
+	shift
+	for e in "$@"; do
+		[[ "$e" == "${element}" ]] && return 0
 	done
 	return 1
```

Comment:
This just works great, but i had thought of something else and in a few tests i did, i found out that comparing the element to the full string of subscripts is a bit faster (well, much faster if the string/array is as big as the ones i used in the tests). To get the full advantage here, the sum of the array subscripts should be passed with the asterisk instead of "@", as in `_bash-it-array-contains-element something ${arr[*]}`. I'm not quite sure why is that, but i'm guessing that it's because with `@` word splitting is invoked, while with `*` it is not, i'm not sure though. So, if you think that's a good idea, it would also be good to replace the arrays in all occurrences of this function to use the asterisk. Not that there'll be some tremendous improvement, but why not.
```suggestion
	local element="${1?}"
	shift
	[[ " $* " == *" $element "* ]]
```

Reply:
No, we specifically make sure to *always* use the `@` specifically so that we do not just munge the words together. Loops are definitely a sore spot for _Bash_ performance, but I don't know of a way to avoid a loop here.

(Think of the case where an array element contains spaces of it's own.)

Reply:
This would also match subscripts with embedded spaces, but I guess your concern is that it would match passed strings in general, not caring whether they're one subscript or multiple ones. I hadn't considered this tbh. No problem though, the loop isn't that slow either 😁

Reply:
The real concern is that it would match *substrings*. If one array item was "some stuff", and we searched for "some" or "stuff", the check would report that the item is already present in the array!